linera-core/src/updater.rs

```diff
@@ -633,23 +629,23 @@ where
         let info = if let Ok(height) = target_block_height.try_sub_one() {
             // Figure out which certificates this validator is missing. In many cases, it's just the
             // last one, so we optimistically send that one right away.
-            let certificate = if let Some(cert) = latest_certificate {
-                cert
+            let hash = self
+                .client
+                .local_node
+                .chain_state_view(chain_id)
+                .await?
+                .block_hashes([height])
+                .await?
+                .into_iter()
+                .next()
+                .ok_or_else(|| {
+                    ChainClientError::InternalError(
+                        "send_chain_information called with invalid target_block_height",
+                    )
+                })?;
```

Comment:
No no, we DON'T want to do this if `latest_certificate` exists. Otherwise we're wasting time again - that was the whole point of the change. This adds 100ms+ to every call.